### PR TITLE
Persistent selected/visibility value for line graph's datasets

### LINKED ISSUE #3272: The datasets selection in the graphs should be persistent
The player should be able to manually determine the datasets which are shown in the graphs and they should remain persistent during the game.

### PR DESCRIPTION
**Pull request by Kasterisk**
**Brief Description of What This PR Does**

The datasets selected to be shown/hidden for a line graph can now be stored and loaded throughout the entirety of the game. This is done using a static variable in LineGraph to track the saved values.

**Related Issues**

Closes #3272 

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/gui_common/charts/line/LineChart.cs [hunk 1]
```diff
@@ -377,14 +384,18 @@ public void ClearDataSets()
...
-            else if (visibleDataSetCount < initialVisibleDataSets && data.Key != defaultDataSet)
-            {
-                visibleDataSetCount++;
+                var visible = visibleDataSetCount < initialVisibleDataSets;
+
+                // Override visible value if stored value exists
+                if (stored.TryGetValue(data.Key, out bool value))
+                    visible = value;
+
+                GD.Print(UpdateDataSetVisibility(data.Key, visible));
```
Comment (hhyyrylainen) [inline src/gui_common/charts/line/LineChart.cs:395]: Accidentally left print here?

Comment (Kasterisk) [reply src/gui_common/charts/line/LineChart.cs:395]: Oops.. yeah left over debugging.

**Comment by hhyyrylainen** [comment]
Initially look on the code looks fine now, I'll wait on playtesting results.

**Comment by RacerBG** [comment]
Everything works as expected, thank you again. :smiley:

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**